### PR TITLE
Flv test updates to create performance data results

### PR DESCRIPTION
**Pull request by tjtelan**
Large refactoring to support data collection during testing.

Creating charts and CSVs for the following:
* Producer/Consumer/E2E latency (+ avg and several percentiles)
* Producer/Consumer throughput
* System memory and CPU usage

Basic Pulsar and Kafka client support

Updates to README

### CODE REVIEW

File: tests/runner/Cargo.toml [hunk 1]
```diff
@@ -14,14 +14,19 @@ async-trait = "0.1.21"
...
 serde_json = "1.0"
+csv = "1.1"
 inventory = "0.1"
 tokio = { version = "1.4", features = ["macros"] }
 bencher = "0.1"
 prettytable-rs = "0.8"
 hdrhistogram = "7.3.0"
 async-lock = "2.4.0"
+async-channel = "1.6.1"
+sysinfo = "0.19.1"
```
Comment (sehz) [inline tests/runner/Cargo.toml:27]: we should create different feature flag. By default, performance run be disabled.  Pulsar and Kafka run should be also made as optional. 

Another thought  is to create separate runner for performance

File: tests/runner/src/main.rs [hunk 2]
```diff
@@ -30,35 +40,38 @@ fn main() {
...
-        let fluvio_client = cluster_setup(&option.environment).await;
+        let fluvio_client = cluster_setup(&option).await;
 
         // Check on test requirements before running the test
-        if !FluvioTestDriver::is_env_acceptable(
-            &(test_meta.requirements)(),
-            &TestCase::new(option.environment.clone(), test_opt.clone()),
-        ) {
-            exit(-1);
+        if &option.environment.cluster_type == "fluvio" {
```
Comment (sehz) [inline tests/runner/src/main.rs:63]: Instead of matching string, map cluster type to enum in the arg

File: tests/runner/src/main.rs [hunk 3]
```diff
@@ -69,37 +82,312 @@ fn main() {
...
+
+            if let Some(l) = panic_info.location() {
+                eprintln!("file: {} @ line {}", l.file(), l.line());
+            } else {
+                eprintln!("There was no location information from panic.");
+            }
         }));
-        */
+
+        let system_profiler_client = fluvio_client.clone();
```
Comment (sehz) [inline tests/runner/src/main.rs:100]: this should be made put under feature flag or make as optional

File: tests/runner/src/main.rs [hunk 4]
```diff
@@ -69,37 +82,312 @@ fn main() {
...
+        let (s, r) = async_channel::unbounded();
+        spawn(async move {
+            let mut s = System::new_all();
```
Comment (sehz) [inline tests/runner/src/main.rs:103]: move this into separate module or async function to make this more modular

File: tests/runner/src/main.rs [hunk 5]
```diff
@@ -69,37 +82,312 @@ fn main() {
...
-            fluvio_client,
+            fluvio_client.clone(),
+            s,
         )
         .await;
-        cluster_cleanup(option.environment).await;
 
-        println!("{}", test_result)
+        // TODO: Support some form of cluster cleanup for other clusters
+        if &option.environment.cluster_type == "fluvio" {
```
Comment (sehz) [inline tests/runner/src/main.rs:155]: use environment driver rathe than hard code this

File: tests/runner/src/main.rs [hunk 6]
```diff
@@ -69,37 +82,312 @@ fn main() {
...
+            cluster_cleanup(option.environment).await;
+        }
+
+        if let Ok(t) = test_result {
+            println!("{}", t);
+
+            if !option.runner_opts.skip_data_save {
```
Comment (sehz) [inline tests/runner/src/main.rs:162]: move this into chart module

File: tests/runner/src/main.rs [hunk 7]
```diff
@@ -111,36 +399,53 @@ async fn cluster_cleanup(option: EnvironmentSetup) {
...
-        num_consumers: 0,
-        bytes_produced: 0,
-        bytes_consumed: 0,
-        produce_latency: Histogram::<u64>::new_with_bounds(1, u64::MAX, 2).unwrap(),
-        consume_latency: Histogram::<u64>::new_with_bounds(1, u64::MAX, 2).unwrap(),
-        topic_create_latency: Histogram::<u64>::new_with_bounds(1, u64::MAX, 2).unwrap(),
-    }))
+async fn cluster_setup(option: &BaseCli) -> Arc<RwLock<TestDriver>> {
+    // TODO: Maybe have an enum for the types of cluster drivers we support
+    match option.environment.cluster_type.as_str() {
```
Comment (sehz) [inline tests/runner/src/main.rs:404]: move this under environment driver rather than hard coding them

File: tests/runner/src/main.rs [hunk 8]
```diff
@@ -111,36 +399,53 @@ async fn cluster_cleanup(option: EnvironmentSetup) {
...
+
+            Arc::new(RwLock::new(
+                TestDriver::new(fluvio_client)
+                    .with_cluster_addr(option.environment.cluster_addr.clone())
+                    .with_producer_batch_ms(option.environment.batch_ms)
+                    .with_producer_batch_size(option.environment.batch_kbytes)
+                    .with_producer_record_size(option.environment.message_size),
+            ))
+        }
+        "pulsar" => Arc::new(RwLock::new(
```
Comment (sehz) [inline tests/runner/src/main.rs:433]: pulsar and kafka should under feature flag

**Review by sehz — CHANGES_REQUESTED** [review]
This too big a PR.  We should break this into a smaller PR:

**Comment by tjtelan** [comment]
@sehz Sure, I can break this up into smaller pieces. I was just hoping that we're now able to commit to the data it produces currently? 

If we're happy with the numbers that we're collecting (like good enough for publishing in a blog post) then there shouldn't be any more changes needed other than what you've requested.

**Comment by sehz** [comment]
That discussion is out of scope for this PR

**Comment by tjtelan** [comment]
Closing in favor of #1380 - I'll bring this branch into the repo with split up PRs

**Pull request closed without merging**